doc/move_group_interface/src/move_group_interface_tutorial.cpp

```diff
@@ -32,169 +32,456 @@
+    // a scaling factor of the maxiumum speed of each joint. Note this is the joint velocities
+    // not the end effector velocities.
+    move_group_.setMaxVelocityScalingFactor(0.1);
+
+    // We want the Cartesian path to be interpolated at a resolution of 1 cm
+    // which is why we will specify 0.01 as the max step in Cartesian
+    // translation.  We will specify the jump threshold as 0.0, effectively disabling it.
+    // Warning - disabling the jump threshold while operating real hardware can cause
+    // large unpredictable motions of redundant joints and could be a safety issue.
+    moveit_msgs::RobotTrajectory trajectory;
+    const double jump_threshold = 0.0;
+    const double eef_step = 0.01;
+    double fraction = move_group_.computeCartesianPath(target_poses, eef_step, jump_threshold, trajectory);
+
+    // While we didn't use the planner to create this trajectory it still can be added to a plan
+    plan.trajectory_ = trajectory;
+    // Moving the robot with this plan is the same as before.
+    //
+    // When finished remember to reset the scalingfactor to return to normal speeds.
+    move_group_.setMaxVelocityScalingFactor(1);
+    // END_SUB_TUTORIAL
+
+    ROS_INFO_NAMED("tutorial", "Visualizing plan (Cartesian path) (%.2f%% acheived)", fraction * 100.0);
+    return fraction;
+  }
+
+  /**
+  * Adds the given object to the world
+  * shape_msgs::SolidPrimitive primitive - The shape
+  * geometry_msgs::Pose box_pose - The location
+  * std::string name - The object ID
+  **/
+  void createObject(const shape_msgs::SolidPrimitive& primitive, const geometry_msgs::Pose& box_pose, std::string name)
+  {
+    // BEGIN_SUB_TUTORIAL add_obj
+    // Create a collisionObject.
+    moveit_msgs::CollisionObject collision_object;
+    collision_object.header.frame_id = move_group_.getPlanningFrame();
+    collision_object.id = name;
+    collision_object.primitives.push_back(primitive);
+    collision_object.primitive_poses.push_back(box_pose);
+    collision_object.operation = collision_object.ADD;
+
+    // Add it to our storage.
+    collision_objects_.push_back(collision_object);
+    object_ids_.push_back(collision_object.id);
+
+    // Now, let's add the collision object into the world.
+    planning_scene_interface_.addCollisionObjects(collision_objects_);
+    // Once added new plans will avoid the object.
+    // END_SUB_TUTORIAL
+  }
+
+  /**
+  * Removes all objects from the world
+  **/
+  void removeObjects()
+  {
+    // BEGIN_SUB_TUTORIAL remove_obj
+    //
+    // To remove the object from the planning scene.
+    planning_scene_interface_.removeCollisionObjects(object_ids_);
+    // To remove from our storage.
+    object_ids_.clear();
+    collision_objects_.clear();
+    // END_SUB_TUTORIAL
+  }
+
+  /**
+  * Attaches a object to the robot
+  * std::string id - ID of the object to attach
+  **/
+  void attachObject(std::string id)
```

Comment:
why are we wrapping single function calls with a function? this function provides no value other than adding complexity to what should be a simple tutorial